lib/runner/extensions/waterfall.command.js

```diff
@@ -51,11 +51,13 @@ module.exports = {
     init: function (done) {
         var state = this.state;
 
-        // ensure that the environment and globals are in VariableScope instance format
+        // ensure that the environment, globals and collectionVariables are in VariableScope instance format
         state.environment = VariableScope.isVariableScope(state.environment) ? state.environment :
             new VariableScope(state.environment);
         state.globals = VariableScope.isVariableScope(state.globals) ? state.globals :
             new VariableScope(state.globals);
+        state.collectionVariables = VariableScope.isVariableScope(state.collectionVariables) ?
```

Comment:
Could you add a `@todo` to move this to item level, so we can support `Item` and `ItemGroup` variables?

Reply:
Wrong place.